packages/sdk-react/src/core/hooks/useAutosizeGridColumns.ts

```diff
@@ -0,0 +1,137 @@
+  const serializationKey = 'Monite-DataGridColumns-' + columnSerializationKey;
+
+  // Autosize columns
+  useLayoutEffect(() => {
+    const grid = gridApiRef.current;
+    const timeouts: number[] = [];
+    if (
+      rows && // Do not autosize until rows are loaded
+      rows.length &&
+      grid &&
+      grid.autosizeColumns && // Will be undefined in test environment
+      grid.getRowsCount() && // Make sure the grid was already rendered so that autosizeColumns function will work
+      grid.getCellElement(rows[0].id, grid.getAllColumns()[0].field) && // Make sure the grid was already rendered so that autosizeColumns function will work
+      !areCounterpartsLoading && // Skip autosize until counterparts are loaded
+      !autoSizePerformed.current && // Do not perform autosize twice
+      !localStorage.getItem(serializationKey) // Do not perform autosize on next page loads - we preserve column state, set by user
+    ) {
+      timeouts.push(
+        window.setTimeout(() => {
+          flushSync(() => {
+            timeouts.push(
+              window.setTimeout(async () => {
+                const previousColumnsState = grid.getAllColumns();
+                await grid.autosizeColumns({
+                  includeHeaders: true,
+                  includeOutliers: true,
+                });
+                previousColumnsState.forEach((previousColumnState) => {
+                  const currentColumnState = grid.getColumn(
+                    previousColumnState.field
+                  );
+                  if (
+                    currentColumnState?.width &&
+                    previousColumnState?.width &&
+                    (currentColumnState.width < previousColumnState.width ||
+                      currentColumnState.width > 600)
+                  ) {
+                    grid.setColumnWidth(
+                      previousColumnState.field,
+                      previousColumnState.width
+                    );
+                  }
+                });
+
+                // Some Payables may have a very long document numbers.
+                // Limit width of the document_id column
+                const documentColumn = grid.getColumn('document_id');
+                if (
+                  documentColumn &&
+                  (documentColumn?.width || 0) > maximumDocumentIdColumnWidth
+                ) {
+                  grid.setColumnWidth(
+                    'document_id',
+                    maximumDocumentIdColumnWidth
+                  );
+                }
+
+                autoSizePerformed.current = true;
+              }, 1)
+            );
+          });
+        }, 1)
+      );
+    }
+
+    return () => {
+      timeouts.forEach(window.clearTimeout); // Cancel pending timeouts on unmount
+    };
+  }, [gridApiRef, rows, columns, areCounterpartsLoading, serializationKey]);
+
+  // Save columns width when switching between pages
+  useLayoutEffect(() => {
```

Comment:
Consider to add early return for trivial cases ` if (!rows || !gridApiRef.current) return; `

Reply:
Checking for rows count makes no sense for deserialization of column's width. I added check for columns number though.